### PR TITLE
Verlet integration

### PR DESCRIPTION
**Pull request by lashlockbhs**
Problems:
Circles clump together,
Vel after collison is wrong,
Vel after collison is not based off mass,
If you want to fix please make a another branch and make a pull called "vertlet fix".

### CODE REVIEW

**Comment by lashlockbhs** [comment]
Mr. Seibel please review my collison code and compare it to yours, please let me know whats up with it.

File: projects/circlePhysics/public/script.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ import {
   setCanvas,
   drawFilledCircle,
   clear,
-  width,
-  height,
+  //width,
```
Comment (ionisea) [inline projects/circlePhysics/public/script.js:6]: ???

File: projects/circlePhysics/public/script.js [hunk 2]
```diff
@@ -30,91 +30,138 @@ import {
   shapeArea,
   vector,
   twoPointXYDif,
-  twoShapeGrav, 
+  twoShapeGrav,
+  findDerivative,
 } from './math.js';
-
+const width = 100;
```
Comment (ionisea) [inline projects/circlePhysics/public/script.js:36]: ah

**Review by ionisea — COMMENTED** [review]
change the dang var names to be more clear!

File: projects/circlePhysics/public/script.js [hunk 3]
```diff
@@ -30,91 +30,138 @@ import {
...
+        shape1.pOld.x = shape1.pCurr.x;
+        shape1.pOld.y = shape1.pCurr.y;
+
+        shape2.pOld.x = shape2.pCurr.x;
+        shape2.pOld.y = shape2.pCurr.y;
+
+        shape1.pCurr.y += Math.sin(angle) * ((overLap +2) / 2) * (0.3);
+        shape1.pCurr.x += Math.cos(angle) * ((overLap+2) / 2) * (0.3);
+        shape2.pCurr.y += Math.sin(angle2) * ((overLap+2) / 2) * (0.3);
+        shape2.pCurr.x += Math.cos(angle2) * ((overLap+2) / 2) * (0.3);
```
Comment (gigamonkey) [inline projects/circlePhysics/public/script.js:151]: I feel like these shouldn't all be moving in a positive direction.

**Review by gigamonkey — APPROVED** [review]
Left one comment on your collision code but the main thing I notice is there are a lot of changes here that are also in @ionisea's latest PR. You all need to work on getting smaller bits integrated more frequently so you can build on each other's work in a reasonable way.

File: projects/circlePhysics/public/script.js (hunk 3 above)
Comment (lashlockbhs) [reply projects/circlePhysics/public/script.js:151]: We reviewed in class and turns out this is not the problem

Comment (gigamonkey) [reply projects/circlePhysics/public/script.js:151]: Okay, as we figured out I was wrong about this. I didn't notice that two of these were using `angle` and the other `angle2`.

Comment (lashlockbhs) [reply projects/circlePhysics/public/script.js:151]: The problem we have determined is with the overlap calcualtions.

**Comment by lashlockbhs** [comment]
Can we try to get this merged today?

**Review by ionisea — APPROVED** [review]
there's a few big problems but at least verlet works

**Pull request merged**